src/app/components/home/home.component.ts

```diff
@@ -5,4 +5,30 @@ import { Component } from '@angular/core';
   templateUrl: './app/components/home/home.component.html',
   styleUrls: ['./app/components/home/home.component.css']
 })
-export class HomeComponent { }
\ No newline at end of file
+export class HomeComponent {
+
+
+// var myIndex = 0;
+
+
+carousel() {
+  let myIndex: number = 0;
+  console.log("test");
+    // let i: number;
+    let slides = document.getElementsByClassName("mySlides");
+    console.log(slides);
+
+    for (let slide in slides) {
```

Comment:
Here is an error. It should be "for (let slide of slides)".
Please don't commit unfinished code and merge into Master, there're tons of comment in home.component.ts.